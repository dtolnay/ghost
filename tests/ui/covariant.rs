#![feature(generic_const_items)]
#![allow(incomplete_features)]

use ghost::phantom;

#[phantom]
struct CovariantPhantom<T>;

fn require_covariant<'a>(phantom: CovariantPhantom<&'static str>) -> CovariantPhantom<&'a str> {
    phantom
}

fn require_contravariant<'a>(phantom: CovariantPhantom<&'a str>) -> CovariantPhantom<&'static str> {
    phantom
}

fn main() {}
