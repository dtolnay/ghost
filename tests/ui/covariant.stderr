error: lifetime may not live long enough
  --> tests/ui/covariant.rs:14:5
   |
13 | fn require_contravariant<'a>(phantom: CovariantPhantom<&'a str>) -> CovariantPhantom<&'static str> {
   |                          -- lifetime `'a` defined here
14 |     phantom
   |     ^^^^^^^ returning this value requires that `'a` must outlive `'static`
